cmd/juju/application/bundle.go

```diff
@@ -1120,7 +1123,20 @@ func (h *bundleHandler) consumeOffer(change *bundlechanges.ConsumeOfferChange) e
 	if url.Source == "" {
 		url.Source = h.controllerName
 	}
-	consumeDetails, err := h.api.GetConsumeDetails(url.AsLocal().String())
+	// Get the consume details from the offer api. We don't use the generic
+	// DeployAPI as we may have to contact another controller to gain access
+	// to that information.
+	controllerOfferAPI, err := h.getConsumeDetailsAPI(url)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	defer controllerOfferAPI.Close()
```

Comment:
Small nit-pick: can you change this to `defer func(){ _ = controllerOfferAPI.Close() }()`. It's a bit more verbose but clearly indicates that we don't care about the error.

Reply:
following @manadart :dagger: